src/main/sensors/gyro.c

```diff
@@ -410,35 +412,43 @@ static FAST_CODE void gyroUpdateSensor(gyroSensor_t *gyroSensor)
 
 FAST_CODE void gyroUpdate(void)
 {
```

Comment:
Maybe iterate over all gyros (including single-gyro case) and accumulate ADC + count, then calculate result. 
```
int adc[3] = {0,0,0};
for(i...)
  if (gyro.gyroToUse == i || gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_ALL) {
   if (calibrated(i)) {
     adc[x] +=  gyro.gyroSensor[i].gyroDev.gyroADC.x * gyro.gyroSensor[i].gyroDev.scale;
     active++;